src/utils/breaches.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  getBreachesForEmail,
+  getFilteredBreaches,
+  HibpLikeDbBreach,
+} from "./hibp";
+import { getSha1 } from "./fxa";
+import { captureMessage } from "@sentry/node";
+import { EmailAddressRow, SubscriberRow } from "knex/types/tables";
+
+export type BundledVerifiedEmails = {
+  email: string;
+  breaches: HibpLikeDbBreach[];
+  id: number;
+  primary: boolean;
+  verified: boolean;
+  hasNewBreaches?: number;
+};
+
+export type AllEmailsAndBreaches = {
+  unverifiedEmails: EmailAddressRow[];
+  verifiedEmails: BundledVerifiedEmails[];
+};
+
+type userType =
+  | ({
+      email_addresses: Array<{
+        id: EmailAddressRow["id"];
+        email: EmailAddressRow["email"];
+      }>;
+    } & SubscriberRow)
+  | undefined;
+
+async function getAllEmailsAndBreaches(
+  user: userType,
+  allBreaches: HibpLikeDbBreach[],
+): Promise<AllEmailsAndBreaches> {
+  // @ts-ignore: function will be deprecated
+  const verifiedEmails: BundledVerifiedEmails[] = [];
+  // @ts-ignore: function will be deprecated
+  const unverifiedEmails: EmailAddressRow[] = [];
+
+  if (!user) {
+    const errMsg = "getAllEmailsAndBreaches: subscriber cannot be undefined";
+    console.error(errMsg);
+    captureMessage(errMsg);
+
+    // @ts-ignore: function will be deprecated
+    return { verifiedEmails, unverifiedEmails };
+  }
+  if (!allBreaches || allBreaches.length === 0) {
+    const errMsg =
+      "getAllEmailsAndBreaches: allBreaches object cannot be empty";
+    console.error(errMsg);
+    captureMessage(errMsg);
+    // @ts-ignore: function will be deprecated
+    return { verifiedEmails, unverifiedEmails };
+  }
+
+  const monitoredEmails = await getUserEmails(user.id);
+  verifiedEmails.push(
+    await bundleVerifiedEmails({
+      user,
+      email: user.primary_email,
+      recordId: user.id,
+      recordVerified: user.primary_verified,
+      allBreaches,
+    }),
+  );
+  for (const email of monitoredEmails) {
+    if (email.verified) {
+      verifiedEmails.push(
+        await bundleVerifiedEmails({
+          user,
+          email: user.primary_email,
+          recordId: email.id,
+          recordVerified: email.verified,
+          allBreaches,
+        }),
+      );
+    } else {
+      unverifiedEmails.push(email);
+    }
+  }
+
+  // get new breaches since last shown
+  for (const emailEntry of verifiedEmails) {
+    // /** @type {any[]} */
+    const newBreachesForEmail = emailEntry.breaches.filter(
+      (breach) => breach.AddedDate >= user.breaches_last_shown,
+    );
+
+    for (const newBreachForEmail of newBreachesForEmail) {
+      newBreachForEmail.NewBreach = true; // add "NewBreach" property to the new breach.
+      emailEntry.hasNewBreaches = newBreachesForEmail.length; // add the number of new breaches to the email
+    }
+  }
+
+  return { verifiedEmails, unverifiedEmails };
+}
+
+function addRecencyIndex(foundBreaches: HibpLikeDbBreach[]) {
+  // /**
+  //  * @type {any[]}
```

Comment:
Same question re: this type definition - we were using these JSDoc type annotations to get a little type checking for JS files before we fully switched to typescript but we shouldn't need them anymore.